### ISSUE TITLE
Model Distillation

### ISSUE THREAD

**Issue by nhantien** [Capstone-AutoML/AutoML_Capstone#61]
### Summary

There doesn’t appear to be a straightforward, ready-to-use implementation for performing model distillation with YOLOv8. Most existing distillation libraries (e.g., TorchDistill, MMDetection) require significant custom integration to work with YOLOv8's architecture.

### Current Understanding

I'm currently researching how to implement knowledge distillation manually using YOLOv8 models. My preliminary understanding is as follows:

* We have a **teacher model** — a larger, pretrained YOLOv8 model (e.g., `yolov8-large`).
* We also have a **student model** — a smaller YOLOv8 model (e.g., `yolov8-small`) that we want to train using both ground truth labels and guidance from the teacher.
* The training process involves **two loss functions**:

  1. A standard detection loss between the **student's predictions and the ground truth**.
  2. A distillation loss between the **student's predictions and the teacher's predictions**.
* These two losses are **combined during training**, for example:

  ```python
  combined_loss = alpha * loss_student_truth + (1 - alpha) * loss_teacher_student
  ```

## Some results so far

- Ultralytics discussion about distillation: https://github.com/ultralytics/ultralytics/issues/2858
- A person implements the distillation process, but did not release as a pip package, also no apparent documentation on how to work with ithttps://github.com/danielsyahputra/yolo-distiller
- https://github.com/wonbeomjang/yolov5-knowledge-distillation
- Generic pytorch distillation package https://github.com/yoshitomo-matsubara/torchdistill
- Of course, a youtube video https://www.youtube.com/watch?v=gADXP5daZeM&t=228s&ab_channel=DinguSagar
- Pytorch reading: https://docs.pytorch.org/tutorials/beginner/knowledge_distillation_tutorial.html
